src/SAWScript/Crucible/LLVM/Builtins.hs

```diff
@@ -746,33 +750,38 @@ verifyPrestate opts cc mspec globals =
 assumptionsContainContradiction ::
   (Crucible.HasPtrWidth (Crucible.ArchWidth arch), Crucible.HasLLVMAnn Sym) =>
   LLVMCrucibleContext arch ->
-  ProofScript SatResult ->
+  ProofScript () ->
   [Crucible.LabeledPred Term Crucible.AssumptionReason] ->
   TopLevel Bool
 assumptionsContainContradiction cc tactic assumptions =
   do
+     let sym = cc^.ccBackend
+     st <- io $ Common.sawCoreState sym
+     let sc  = saw_ctx st
      pgl <- io $
       do
-         let sym = cc^.ccBackend
-         st <- Common.sawCoreState sym
-         let sc  = saw_ctx st
          -- conjunction of all assumptions
          assume <- scAndList sc (toListOf (folded . Crucible.labeledPred) assumptions)
          -- implies falsehood
          goal  <- scImplies sc assume =<< toSC sym st (W4.falsePred sym)
-         goal' <- predicateToProp sc Universal goal
+         goal' <- boolToProp sc [] goal
          return $ ProofGoal 0 "vc" "vacuousness check" goal'
-     evalStateT tactic (startProof pgl) >>= \case
-       Unsat _stats -> return True
-       SatMulti _stats _vals -> return False
+     res <- runProofScript tactic pgl
+     case res of
+       ValidProof _ _     -> return True
+       InvalidProof _ _ _ -> return False
+       UnfinishedProof _  ->
+         -- TODO? is this the right behavior?
+         do printOutLnTop Warn "Could not determine if preconditions are vacuous"
```

Comment:
It seems like reasonable behavior to me, at least unless we see evidence to the contrary.